services/checks/checks.go

```diff
@@ -578,23 +590,63 @@ func (s *Service) processResults(ctx context.Context, check check.Check, target
 		return nil, err
 	}
 
-	env, err := starlark.NewEnv(check.Name, check.Script, funcs)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to prepare starlark environment")
-	}
-
 	l := s.l.WithFields(logrus.Fields{
 		"name":       check.Name,
 		"id":         resID,
 		"service_id": target.serviceID,
 	})
+
+	scriptInput, err := agentpb.MarshalActionQueryDocsResult(r)
+	if err != nil {
+		return nil, err
+	}
+
+	input := &StarlarkScriptData{
+		CheckName:   check.Name,
+		Script:      check.Script,
+		Funcs:       funcs,
+		PrintFn:     l.Debugln,
+		ScriptInput: scriptInput,
+	}
+
+	process := exec.Command("./bin/pmm-managed-starlark")
+	process.SysProcAttr = &syscall.SysProcAttr{
+		Pdeathsig: syscall.SIGTERM,
+	}
+	_ = unix.Setrlimit(unix.RLIMIT_CPU, &unix.Rlimit{Cur: 4, Max: 4})
+	_ = unix.Setrlimit(unix.RLIMIT_DATA, &unix.Rlimit{Cur: 100000000, Max: 100000000})
+
+	pipe, err := process.StdinPipe()
+	defer pipe.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	encoder := gob.NewEncoder(pipe)
```

Comment:
pmm-agent sends protobuf-encoded data to pmm-managed; just pass the same data there. No need to use gob or anything else

Reply:
pmm-agent only gives us the input for the scripts but I need other data too like the check name, script body, funcs, etc. 

This is what I pass to the binary 
```go
type StarlarkScriptData struct {
	CheckName   string            
	Script      string                    `
	Funcs       map[string]starlark.GoFunc 
	PrintFn     starlark.PrintFunc         
	ScriptInput []map[string]interface{}   
}
```

and for that I need `gob` because `json` can't encode types like `starlark.GoFunc `

Reply:
gob can't actually encode Go code either. And why you want to pass it via stdin instead of initialising it in the binary itself?

Reply:
by initializing in the binary I think you mean `starlark.NewEnv` which I already do, but the arguments to this function and `env.Run` are in `checks.go` which I need to supply to the binary

Reply:
Yeah, so want prevents you from importing that package in pmm-managed-starlark and using it?

Reply:
moved some function calls to the binary but I still need to pass some extra data like the check name, version and the script which cannot be available in the binary by importing a package and calling a function since these are arguments to `processResults` for that I have to encode them in a json struct and send it over using STDIN